### PR TITLE
Update test harness and handling .reference files

### PR DESCRIPTION
**Pull request by lqd**
This PR fixes (hopefully) #139:
- by introducing control over colors when formatting errors
- making the test harness not use colors when formatting diagnostics: the very readable result is visible in the "bless colorless test expectations" commit.
- if I understood #139 correctly: all .ref files were to be "ignored" during tests (except when blessing, where they are still generated), removed from the repo, and `gitignore`d ? I hope this understanding is correct, as the last few commits do that.

### CODE REVIEW

File: components/dada-lang/src/test_harness.rs [hunk 1]
```diff
@@ -166,35 +166,31 @@ impl Options {
             &expected_diagnostics.compile,
             &mut errors,
         )?;
-        self.check_output_against_ref_file(
+        self.maybe_bless_ref_file(
```
Comment (nikomatsakis) [inline components/dada-lang/src/test_harness.rs:169]: I was going to suggest that we make this *always* bless, but maybe it's good to have a mode that doesn't generate output files. (I somewhat wonder if `--bless` should be the default, even, and instead we have `--check`, but I guess that could be surprising.)

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
Thanks @lqd!

bors r+

File: components/dada-lang/src/test_harness.rs (hunk 1 above)
Comment (lqd) [reply components/dada-lang/src/test_harness.rs:169]: Could be a good idea to always bless by default and ensure you're not looking at old files.

Comment (nikomatsakis) [reply components/dada-lang/src/test_harness.rs:169]: If you wanted to open up a follow-up PR, would be welcome. Maybe a good idea would be to *try* and update the ref files, but if an I/O error occurs, just *warn* about it and don't fail the tes.t

**Pull request merged**